### ISSUE TITLE
Scrolling up very choppy after using scrollToIndex

### ISSUE THREAD

**Issue by manuhook** [Shopify/flash-list#582]
Hi, thank you for this great lib !
I would like to replace Flatlist with it on our app but I'm facing an annoying bug with scrollToIndex.

## Current behavior

On Android or iOS, when using **scrollToIndex** on large list (around 500 items), once the scrollToIndex is done, when I scroll up, the scroll is very choppy. scrolling dow is fine.

iOS : iPhone 12 Pro max : https://imgur.com/a/kf5zvBC
Android : Samsung Galaxy S7 : https://imgur.com/a/fwBBlUt

## Expected behavior

Scrolling up should be fluid.

## To Reproduce

this.listRef.scrollToIndex({animated:false, index: index})

Adding a setimeout is not helping.

## Platform:

- [X] iOS
- [X] Android

## Environment

"@shopify/flash-list": "^1.2.1"

Thank you for your help.

**Comment by naqvitalha** [comment]
Does it work fine with FlatList? If yes, can you share a video?

I understand what's going on but I also want to understand if this works with FlatList.

**Comment by manuhook** [comment]
Yes it's working with Flatlist but it's much slower (40 secondes on the Galaxy S7)

Here a video on the iPhone 12 Pro with Flatlist : https://www.cookomix.com/temp/RPReplay_Final1662112584.mp4

**Comment by manuhook** [comment]
Looks like I have the same behavior if I use initialScrollIndex

**Comment by oiver555** [comment]
Any fix or workaround on this? I'm having the same issue, and this is only thing holding me back from making the switch to FlashList which is something I really want to do. 

@manuhook Were you able to find a workaround?

**Comment by manuhook** [comment]
> Any fix or workaround on this? Or better yet if anyone can give an explanation as to why this is happening would be helpful. I'm having the same issue, and this is only thing holding me back from making the switch to FlashList which is something I really want to do.
> 
> @manuhook Were you able to find a workaround?

Nop I didn't find a workaround with FlashList, I changed my approach by still using FlatList with this library to avoid the problem when a scroll to index is needed : https://github.com/GetStream/react-native-bidirectional-infinite-scroll

**Comment by RalissonMattias** [comment]
Oh, same problem here

**Comment by joaotmachado** [comment]
Same here

**Comment by joesermon** [comment]
Same problem @naqvitalha any news on this issue please ?

**Comment by irisjae** [comment]
I've published an analysis of this problem and a patch [here](https://github.com/irisjae/recyclerlistview/edit/master/README.md); feel free to test it out.

**Comment by irisjae** [comment]
Glad you find the effort useful! Hopefully we can get some maintainers to chime in too.

**Comment by irisjae** [comment]
@SergioAN007 Not without further work on the patch, unfortunately.

This patch works by reimplementing the layout algorithm of flash-list/recyclerlistview (which is separately implemented in the Javascript layouting, and also natively in Swift and Kotlin). The algorithm itself is not algorithmically complex and even fairly short, but has a few edge cases to look out for in testing on real devices. Since I did not have an app that used multiple columns, I did not have a good test case to validate a multiple column implementation and decided to simplify it that way.

If you'd like to work on this, I'm happy to provide any help. If the maintainer decides to work on merging my patch, I probably will also be motivated to look into it, so you might also consider voicing your support to help make that happen.

**Comment by bolandrm** [comment]
We're also unable to use FlashList due to this issue.
